### PR TITLE
Fix/click and collect

### PR DESCRIPTION
**Pull request by kuchichan**
I want to merge this change because it fixes quite harsh error, when no shipping address has been passed to checkout query (in `availableCollectionPoints` resolver). Although country code is needed in order to calculate availableCollectionPoints, we can fallback to channel, because in most cases passing entire `shipping_address` for C&C would be superfluous.  What is more, country are now used in `populatedb` for generated channels.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/checkout/types.py [hunk 1]
```diff
@@ -491,20 +491,27 @@ def apply_price_to_shipping_method(channel_listings):
...
     def resolve_available_collection_points(root: models.Checkout, info):
         def get_available_collection_points(data):
-            address, lines, checkout_info = data
+            address, lines, checkout_info, channel = data
+            country_code = address.country.code if address else None
+
             return get_valid_collection_points_for_checkout(
-                lines, checkout_info, country_code=address.country.code
+                lines,
+                checkout_info,
```
Comment (maarcingebala) [inline saleor/graphql/checkout/types.py:499]: I guess this function doesn't need `checkout_info` anymore.

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 2]
```diff
@@ -1547,6 +1547,23 @@ def test_checkout_avail_collect_points_returns_empty_list_when_not_in_shipping_z
...
+    checkout_with_items_for_cc.shipping_address = None
+    checkout_with_items_for_cc.save()
+
+    variables = {"token": checkout_with_items_for_cc.token}
+    response = api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["checkout"]
+
+    assert data["availableCollectionPoints"] == [
+        {"address": {"streetAddress1": "Tęczowa 7"}, "name": "Local Warehouse"}
```
[the remark this answers is not in the document]
Comment (kuchichan) [reply saleor/graphql/checkout/tests/test_checkout.py]: You're right, done.

File: saleor/graphql/checkout/types.py [hunk 3]
```diff
@@ -491,20 +491,27 @@ def apply_price_to_shipping_method(channel_listings):
     @traced_resolver
     def resolve_available_collection_points(root: models.Checkout, info):
         def get_available_collection_points(data):
-            address, lines, checkout_info = data
+            address, lines, checkout_info, channel = data
+            country_code = address.country.code if address else None
```
[the remark this answers is not in the document]
Comment (kuchichan) [reply saleor/graphql/checkout/types.py]: 👍  Done.

File: saleor/graphql/checkout/types.py (hunk 1 above)
Comment (kuchichan) [reply saleor/graphql/checkout/types.py]: That's true. Done

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]
Looks good, we need to update the changelog as well.

**Pull request merged**